src/Logger/TextFileLogger.php

```diff
@@ -36,27 +36,46 @@
 namespace Infection\Logger;
 
 use function implode;
+use Infection\Mutant\MetricsCalculator;
 use Infection\Mutant\MutantExecutionResult;
+use const PHP_EOL;
 use function Safe\sprintf;
 use function str_repeat;
 use function strlen;
 
 /**
  * @internal
  */
-final class TextFileLogger extends FileLogger
+final class TextFileLogger implements LineMutationTestingResultsLogger
 {
-    protected function getLogLines(): array
+    private $metricsCalculator;
```

Comment:
I wish these loggers would be responsible for the result collection themselves, not MetricsCalculator. Metrics are metrics, logging be logging. Also MetricsCalculator won't collect logs nobody will ask for.

This is probably for another time.

Reply:
I would also like to split the metrics calculator in two. But I got to admit it kind of complicate its consumption for no good reason... Maybe we can discuss that in another issue or in #1099